File: src/main/java/eotree/EOStringData.java
package eotree;

/**
 * EBNF representation:
 * <code>
 * /"[^"]*"/
 * </code>
 */
public class EOStringData extends EOData {
    String str;

    public EOStringData(String str) {
        this.str = str;
    }

    @Override
    public String generateEO(int indent) {
        return "\"" + str + "\"";
    }
}

File: src/main/kotlin/org/polystat/j2eo/eotree/data/EOStringData.kt
package org.polystat.j2eo.eotree.data

/**
 * EBNF representation:
 * `
 * /"[^"]*"/
` *
 */
class EOStringData(
    var str: String,
) : EOData() {
    override fun generateEO(indent: Int): String = indent(indent) + "\"" + str + "\""
}
